From fb257565f4413e495b4f99621d047f323e5d94b2 Mon Sep 17 00:00:00 2001
From: Beka Gozalishvili <beqaprogger@gmail.com>
Date: Wed, 6 Mar 2024 12:55:23 +0400
Subject: [PATCH] Fix compilation on systems using musl libc like alpine-linux

---
 ACE/ace/config-linux.h | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/ACE/ace/config-linux.h b/ACE/ace/config-linux.h
index 954d61f091d..a3f4c5f9904 100644
--- a/ACE/ace/config-linux.h
+++ b/ACE/ace/config-linux.h
@@ -204,6 +204,42 @@
 
 #endif /* __UCLIBC__ */
 
+// To support musl (Unfortunately musl doesn't have a good way to check for musl libc)
+#if !defined (__GLIBC__) && !defined (__UCLIBC__)
+
+#define ACE_HAS_UCONTEXT_T
+#define ACE_HAS_PTHREADS
+#define ACE_HAS_CPU_SET_T
+#define ACE_HAS_SIGINFO_T
+#define ACE_HAS_SOCKLEN_T
+
+#define ACE_LACKS_SIGINFO_H
+#define ACE_LACKS_SYS_SYSCTL_H
+#define ACE_LACKS_RWLOCK_T
+#define ACE_LACKS_ISCTYPE
+#define ACE_LACKS_NETDB_REENTRANT_FUNCTIONS
+
+#  if defined (ACE_SCANDIR_CMP_USES_VOIDPTR)
+#    undef ACE_SCANDIR_CMP_USES_VOIDPTR
+#  endif /* ACE_SCANDIR_CMP_USES_VOIDPTR */
+
+#  if defined (ACE_SCANDIR_CMP_USES_CONST_VOIDPTR)
+#    undef ACE_SCANDIR_CMP_USES_CONST_VOIDPTR
+#  endif /* ACE_SCANDIR_CMP_USES_CONST_VOIDPTR */
+
+#  if defined(__GLIBC__)
+#    undef __GLIBC__
+#  endif /* __GLIBC__ */
+
+#  if defined(ACE_HAS_SEMUN)
+#    undef ACE_HAS_SEMUN
+#  endif /* ACE_HAS_SEMUN */
+
+#define __uint32_t uint32_t
+#define loff_t uint32_t
+
+#endif /* __MUSL__ */
+
 #include /**/ "ace/post.h"
 
 #endif /* ACE_CONFIG_LINUX_H */
-- 
2.43.0.windows.1

